src/importexport/mei/internal/meiconverter.cpp

```diff
@@ -3333,6 +3357,52 @@ std::list<std::string> Convert::getTypeValuesWithPrefix(const std::string& typeS
     return values;
 }
 
+void Convert::layerIdentFromMEI(engraving::EngravingItem* item, const libmei::Element& meiElement)
+{
+    const libmei::AttLayerIdent* layerAtt = dynamic_cast<const libmei::AttLayerIdent*>(&meiElement);
+
+    IF_ASSERT_FAILED(layerAtt) {
+        return;
+    }
+
+    // @color
+    if (layerAtt->HasLayer()) {
+        // without further check we assume the layer to match
+        if (item->hasVoiceAssignmentProperties()) {
+            item->setProperty(engraving::Pid::VOICE_ASSIGNMENT, engraving::VoiceAssignment::CURRENT_VOICE_ONLY);
+        }
+    }
+}
+
+void Convert::layerIdentToMEI(const engraving::EngravingItem* item, libmei::Element& meiElement)
+{
+    libmei::AttLayerIdent* layerAtt = dynamic_cast<libmei::AttLayerIdent*>(&meiElement);
+
+    IF_ASSERT_FAILED(layerAtt) {
+        return;
+    }
+
+    if (item->hasVoiceAssignmentProperties()
+        && (item->getProperty(engraving::Pid::VOICE_ASSIGNMENT).value<engraving::VoiceAssignment>()
+            == engraving::VoiceAssignment::CURRENT_VOICE_ONLY)) {
+        layerAtt->SetLayer(item->voice() + 1);
+    }
+}
+
+void Convert::staffIdentToMEI(const engraving::EngravingItem* item, libmei::Element& meiElement)
+{
+    libmei::AttStaffIdent* staffAtt = dynamic_cast<libmei::AttStaffIdent*>(&meiElement);
+
+    IF_ASSERT_FAILED(staffAtt) {
+        return;
+    }
+
+    libmei::xsdPositiveInteger_List staffList;
+    staffList.push_back(item->staff()->idx() + 1);
+    // TODO: add staff number if centered between staves
```

Comment:
Should the 'discriminating factor' really be whether the number is centered between staves, or should it be when the voice assignment is set to `ALL_VOICE_IN_INSTRUMENT`?
It might be relatively easily to implement using `item->part()->staveIdxList()`.